configure.ac

```diff
@@ -507,8 +507,27 @@ if test x$ac_cv_c_compiler_gnu = xyes ; then
 	MN_ADD_CXXFLAGS([-Wpointer-arith])
 	AC_LANG_POP([C++])
 
-
-	MN_ADD_CFLAGS([-D_FORTIFY_SOURCE=2])
+	dnl some distributions (such as Gentoo) have _FORTIFY_SOURCE always
+	dnl enabled. We test for this situation in order to prevent polluting
+	dnl the console with messages of macro redefinitions.
+	AC_MSG_CHECKING([whether to add -D_FORTIFY_SOURCE=2 to CFLAGS])
+	AC_LINK_IFELSE([
+		AC_LANG_SOURCE(
+			[[
+				int main() {
+				#ifndef _FORTIFY_SOURCE
+					return 0;
+				#else
+					this_is_an_error;
+				#endif
+				}
+			]]
+		)], [
+			AC_MSG_RESULT([yes])
+			MN_ADD_CFLAGS([-D_FORTIFY_SOURCE=2])
+		], [
+			AC_MSG_RESULT([no])
+		])
```

Comment:
@erikd Submitted as https://github.com/peti/autoconf-archive/pull/107
Once its merged, I'll add it here